lib/web/JsonpMainTemplatePlugin.js

```diff
@@ -354,6 +354,34 @@ class JsonpMainTemplatePlugin {
 					return Template.asString([
 						source,
 						"",
+						"var rules = {",
```

Comment:
Such code should only be added if `trustedTypesPolicyName` is set.

User that doesn't want to use `TrustedTypes` shouldn't pay a cost for it.

Reply:
Currently, the `trustedTypesPolicyName` will always be set (either by user or by webpack provided value). 
However, if you want we can make the policy name optional and only include this code if the policy name is set by user. WDYT?

Reply:
@sokra I believe there are two options how load the necessary Trusted Types code conditionally.

1) Check whether Trusted Types are available in global (window) object. If yes, load the code. This would also imply using default value for webpack policy (like we use now). Whether the application has trusted types enforced, enabled or turned off in this implementation isn't a webpack concern.
2) Do not provide a default value for webpack policy and rely on users to set it. Only load the Trusted Types code if the policy name is defined by the user. 

Both option have their +/-.
**The first option -** simplifies Trusted Types integration for users as the only thing they need to do is to add a `webpack` to the CSP Trusted Types header (i.e. it's preconfigured by default by webpack).
**Second option -** Configuration is always necessary by the user. This might make the integration even impossible. If application was created by `create-react-app` (or similar tools) the configuration might not be exposed by default. The benefit is that the code is loaded only when the application really uses Trusted Types.

I think the **1. option** is better because it's simplifies the integration (or makes it possible) and the code reduction provided by 2. option is not a huge benefit. 

WDYT?

Reply:
I agree it would be preferential to make webpack use TT if they are available in the runtime enviroment. The only con is a tiny code size increase if the jsonp template plugin is used. TT are backwards-compatible i.e. nothing breaks if you use them, but they are not enforced. Using them by default makes it easy for webpack users to migrate to Trusted Types (otherwise webpack becomes a blocker, and users must discover how to enable TT support).

@sokra , WDYT?